qml/FileListPage.qml

```diff
@@ -43,48 +57,99 @@ Page {
         model: filteredModel
         currentIndex: -1 // otherwise currentItem will steal focus
 
-        header: SearchPageHeader {
-            id: header
+        header: Item {
+            width: listView.width
+            height: headerContent.height
+        }
+
+        Column {
+            id: headerContent
+
+            parent: listView.headerItem
             width: parent.width
 
-            //: Application title
-            //% "Documents"
-            title: qsTrId("sailfish-office-he-apptitle")
+            PageHeader {
+                //: Application title
+                //% "Documents"
+                title: qsTrId("sailfish-office-he-apptitle")
+            }
+
+            SearchField {
+                id: searchField
 
-            // TODO: uncomment once there are more document sources
-            // title: page.title
+                width: parent.width
+                opacity: page.searchEnabled ? 1.0 : 0.0
+                visible: opacity > 0
 
-            Binding {
-                target: page
-                property: "searchText"
-                value: header.searchText
+                //: Document search field placeholder text
+                //% "Search documents"
+                placeholderText: qsTrId("sailfish-office-tf-search-documents")
+
+                // We prefer lowercase
+                inputMethodHints: Qt.ImhNoAutoUppercase | Qt.ImhPreferLowercase | Qt.ImhNoPredictiveText
+                EnterKey.iconSource: "image://theme/icon-m-enter-close"
+                EnterKey.onClicked: focus = false
+
+                Behavior on opacity { FadeAnimation { duration: 150 } }
+                Behavior on height {
+                    NumberAnimation {
+                        duration: 150
+                        easing.type: Easing.InOutQuad
+                    }
+                }
+                Binding {
+                    target: searchField
+                    when: !searchEnabled
+                    property: "height"
+                    value: 0
+                }
             }
+        }
 
-            Connections {
-                target: menuItemSearch
-                onClicked: header.enableSearch()
+        Connections {
+            target: searchField.activeFocus ? listView : null
+            ignoreUnknownSignals: true
+            onContentYChanged: {
+                if (listView.contentY > (Screen.height / 2)) {
+                    searchField.focus = false
+                }
             }
         }
 
         PullDownMenu {
-            visible: listView.count > 0
```

Comment:
Menu could indeed be omitted if device doesn't have any document files, but having a special case for that didn't seem important enough. As you said, the existing value had to be removed as it was hiding menu during filtering.